063-prometheus-metrics-reporter.md

```diff
@@ -0,0 +1,208 @@
+
+- The `PerSec` suffix is removed by jmx_exporter rules so a few metrics have slightly different names, for example:
+    ```
+    kafka_controller_controllerstats_uncleanleaderelections_total         # jmx_exporter
+    kafka_controller_controllerstats_uncleanleaderelectionspersec_total   # reporter
+
+    kafka_network_requestmetrics_errors_total{request="UpdateMetadata",error="NONE",}        # jmx_exporter
+    kafka_network_requestmetrics_errorspersec_total{request="UpdateMetadata",error="NONE",}  # reporter
+    ```
+    This is due these [mapping rules](https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/metrics/kafka-metrics.yaml#L136-L150). 
+
+- An underscore `_` is added in front of the `percent` suffix by jmx_exporter rules so a few metrics have slightly different names, for example:
+    ```
+    kafka_network_processor_idle_percent{networkProcessor="0",} # jmx_exporter
+    kafka_network_processor_idlepercent{networkProcessor="0",}  # reporter
+    ```
+    This is due these [mapping rules](https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/metrics/kafka-metrics.yaml#L127-L135). 
+
+- The `count` suffix is used instead of `total` for some Counters by jmx_exporter:
+    ```
+    kafka_network_acceptor_acceptorblockedpercent_count{listener="PLAINTEXT",} # jmx_exporter
+    kafka_network_acceptor_acceptorblockedpercent_total{listener="PLAINTEXT",} # reporter
+    ```
+    Prometheus recommends using the `total` suffix and it's actually automatically injected by the Java library. The example jmx_exporter rules correctly replace the suffix in most metrics (like in the metrics mentioned above with `PerSec`) but they don't match all metrics hence the inconsistency.
+
+- Some labels are wrapped twice in quotes by jmx_exporter
+    ```
+    # jmx_exporter
+    kafka_log_logcleanermanager_uncleanable_bytes{logDirectory="\"/tmp/kafka-logs\"",}
+    kafka_log_logcleanermanager_uncleanable_partitions_count{logDirectory="\"/tmp/kafka-logs\"",}
+    kafka_log_logmanager_logdirectoryoffline{logDirectory="\"/tmp/kafka-logs\"",}
+    # reporter
+    kafka_log_logcleanermanager_uncleanable_bytes{logDirectory="/tmp/kafka-logs",}
+    kafka_log_logcleanermanager_uncleanable_partitions_count{logDirectory="/tmp/kafka-logs",}
+    kafka_log_logmanager_logdirectoryoffline{logDirectory="/tmp/kafka-logs",}
+    ```
+    I'm guessing this is done to support paths containing special characters. In the context of Strimzi I don't think this is necessary.
+
+
+In terms of performance, in my very limited testing the reporter is much faster than jmx_exporter. This is not a benchmark or rigorous by any means, I've simply been running `time curl --head http://localhost:{PORT}/metrics` against a single broker on my laptop with both the reporter (on port 8080) and jmx_exporter (on port 9090) running. Since each partition has metrics, I used the number of partition as a proxy to increase the total number of metrics.
```

Comment:
```suggestion
In terms of performance, in my very limited testing the reporter is much faster than jmx_exporter. This is not a benchmark or rigorous by any means, I've simply been running `time curl --head http://localhost:{PORT}/metrics` against a single broker on my laptop with both the reporter (on port 8080) and jmx_exporter (on port 9090) running. Since each partition has metrics, I used the number of partitions as a proxy to increase the total number of metrics.
```